File: 3DS/02/source/one.cpp
#include <stdlib.h>
#include <3ds.h>
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include "utils.h"

using namespace std;

int one()
{
	auto input = readlines("romfs:/input.txt");

	int coords[2] = {0, 0};

	for (auto line : input)
	{
		auto cur = split(line);
		string cmd = cur[0];
		int amount = stoi(cur[1]);

		if (cmd == "forward")
		{
			coords[0] -=- amount;
		}
		else if (cmd == "up")
		{
			coords[1] -=+ amount;
		}
		else if (cmd == "down")
		{
			coords[1] -=- amount;
		}
	}

	return coords[0] * coords[1];
}

File: 02/one.lua
package.path = package.path .. ";../modules/?.lua"
local tools = require "tools"

course = tools.readlines("input.txt")

coords = {0, 0}

for _, mov in pairs(course) do
	cmd, amount = table.unpack(tools.split(mov))
	if cmd == 'forward' then
		coords[1] = coords[1]- -tonumber(amount)
	elseif cmd == 'up' then
		coords[2] = coords[2]+ -tonumber(amount)
	elseif cmd == 'down' then
		coords[2] = coords[2]- -tonumber(amount)
	end
end

print("Product of position and depth: " .. coords[1] * coords[2])
assert(coords[1] * coords[2] == 1804520)
